Python/Product/TestAdapter.Executor/Services/ExecutorService.cs

```diff
@@ -116,6 +114,153 @@ sealed class TestRedirector : Redirector {
             return arguments.ToArray();
         }
 
+        /// <summary>
+        /// Returns true if this is a dry run. Dry runs require a
+        /// &lt;DryRun value="true" /&gt; element under RunSettings/Python.
+        /// </summary>
+        internal static bool IsDryRun(IRunSettings settings) {
+            var doc = Read(settings.SettingsXml);
+            try {
+                var node = doc.CreateNavigator().SelectSingleNode("/RunSettings/Python/DryRun[@value='true']");
+                return node != null;
+            } catch (Exception ex) {
+                Debug.Fail(ex.ToUnhandledExceptionMessage(typeof(TestExecutorUnitTest)));
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Returns true if the console should be shown. This is the default
+        /// unless a &lt;ShowConsole value="false" /&gt; element exists under
+        /// RunSettings/Python.
+        /// </summary>
+        internal static bool ShouldShowConsole(IRunSettings settings) {
+            var doc = Read(settings.SettingsXml);
+            try {
+                var node = doc.CreateNavigator().SelectSingleNode("/RunSettings/Python/ShowConsole[@value='false']");
+                return node == null;
+            } catch (Exception ex) {
+                Debug.Fail(ex.ToUnhandledExceptionMessage(typeof(TestExecutorUnitTest)));
+                return true;
+            }
+        }
+
+        internal static string GetCoveragePath(IEnumerable<TestCase> tests) {
+            string bestFile = null, bestClass = null, bestMethod = null;
+
+            // Try and generate a friendly name for the coverage report.  We use
+            // the filename, class, and method.  We include each one if we're
+            // running from a single filename/class/method.  When we have multiple
+            // we drop the identifying names.  If we have multiple files we
+            // go to the top level directory...  If all else fails we do "pycov".
+            foreach (var test in tests) {
+                string testFile, testClass, testMethod;
+                TestReader.ParseFullyQualifiedTestName(
+                    test.FullyQualifiedName,
+                    out testFile,
+                    out testClass,
```

Comment:
For pytest global test functions, i set the testClass portion of FQN to be the filename.. It would be good to add a testcase for this to make sure it works ok.

Reply:
Added test.